src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.Placement.cs

```diff
@@ -0,0 +1,377 @@
+
+			Point finalPosition;
+
+			switch (placement)
+			{
+				case FlyoutBase.MajorPlacementMode.Top:
+					finalPosition = new Point(
+						x: anchorRect.Left + halfAnchorWidth - halfChildWidth,
+						y: anchorRect.Top - PopupPlacementTargetMargin - desiredSize.Height);
+					break;
+				case FlyoutBase.MajorPlacementMode.Bottom:
+					finalPosition = new Point(
+						x: anchorRect.Left + halfAnchorWidth - halfChildWidth,
+						y: anchorRect.Bottom + PopupPlacementTargetMargin);
+					break;
+				case FlyoutBase.MajorPlacementMode.Left:
+					finalPosition = new Point(
+						x: anchorRect.Left - PopupPlacementTargetMargin - desiredSize.Width,
+						y: anchorRect.Top + halfAnchorHeight - halfChildHeight);
+					break;
+				case FlyoutBase.MajorPlacementMode.Right:
+					finalPosition = new Point(
+						x: anchorRect.Right + PopupPlacementTargetMargin,
+						y: anchorRect.Top + halfAnchorHeight - halfChildHeight);
+					break;
+				case FlyoutBase.MajorPlacementMode.Full:
+#if __IOS__ || __ANDROID__
+					// The status bar should remain visible. On droid, this panel is placed beneath the status bar.
+					desiredSize = new Size(
+						ActualWidth,
+						ActualHeight
+#if __IOS__
+						// On iOS, this panel will cover the status bar, so we have to substract it out.
+						- visibleBounds.Y
+#endif
+					).AtMost(maxSize);
+#else
+					desiredSize = visibleBounds.Size.AtMost(maxSize);
+#endif
+					finalPosition = new Point(
+						x: FindOptimalOffset(desiredSize.Width, visibleBounds.X, visibleBounds.Width, ActualWidth),
+						y: FindOptimalOffset(desiredSize.Height, visibleBounds.Y, visibleBounds.Height, ActualHeight));
+
+					double FindOptimalOffset(double length, double visibleOffset, double visibleLength, double constraint)
+					{
+						// Center the flyout inside the first area that fits: within visible bounds, below status bar, the screen
+						if (visibleLength >= length)
+						{
+							return visibleOffset + (visibleLength - length) / 2;
+						}
+						if (constraint - visibleOffset >= length)
+						{
+							return visibleOffset + ((constraint - visibleOffset) - length) / 2;
+						}
+						else
+						{
+							return (constraint - length) / 2;
+						}
+					}
+					break;
+				default: // Other unsupported placements
+					finalPosition = new Point(
+						x: (visibleBounds.Width - desiredSize.Width) / 2.0,
+						y: (visibleBounds.Height - desiredSize.Height) / 2.0);
+					break;
+			}
+
+			var justification = FlyoutBase.GetJustificationFromPlacementMode(popup.DesiredPlacement, FullPlacementRequested);
+
+			var fits = true;
+			if (!FullPlacementRequested)
+			{
+				if (IsPlacementModeVertical(placement))
+				{
+					var controlXPos = finalPosition.X;
+					fits = TestAndCenterAlignWithinLimits(
+						anchorRect.X,
+						anchorRect.Width,
+						desiredSize.Width,
+						visibleBounds.Left,
+						visibleBounds.Right,
+						justification,
+						ref controlXPos
+					);
```

Comment:
It looks like `TestAndCenterAlignWithinLimits` will always assign a new value to the ref parameter. I think it can be made `out` and the initial value should be unnecessary?

Reply:
Yes indeed, it should have been `out`